cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/instances/RunInstancesAction.java

```diff
@@ -527,8 +538,12 @@
                     .withSecondaryPrivateIpAddressCount(secondaryPrivateIpAddressCount)
                     .build();
 
-            return new QueryApiExecutor().execute(commonInputs, customInputs, ebsInputs, elasticIpInputs, iamInputs,
+            Map<String, String> queryMapResult = new QueryApiExecutor().execute(commonInputs, customInputs, ebsInputs, elasticIpInputs, iamInputs,
                     instanceInputs, networkInputs);
+            if((ReturnCodes.SUCCESS).equals(queryMapResult.get(RETURN_CODE))) {
```

Comment:
@Tirla-Alin Well, now I understood what your intentions are. Great! You can ignore my previous comment about queryMapResult